### PR TITLE
ImageZoom: Fix zoom level not saving

### PR DESCRIPTION
**Pull request by sadan4**

### CODE REVIEW

File: src/plugins/imageZoom/components/Magnifier.tsx [hunk 1]
```diff
@@ -54,6 +54,14 @@ export const Magnifier = ErrorBoundary.wrap<MagnifierProps>(({ instance, size: i
...
     const imageRef = useRef<HTMLImageElement | null>(null);
 
+    useEffect(() => {
+        return () => {
+            if (settings.store.saveZoomValues) {
+                settings.store.zoom = zoom.current;
+                settings.store.size = size.current;
+            }
+        };
+    });
```
Comment (Nuckyz) [inline src/plugins/imageZoom/components/Magnifier.tsx:64]: Shouldnt this effect have an empty dependency array?

**Review by Nuckyz — CHANGES_REQUESTED** [review]

File: src/plugins/imageZoom/components/Magnifier.tsx (hunk 1 above)
Comment (sadan4) [reply src/plugins/imageZoom/components/Magnifier.tsx:64]: isnt nothing the same as an empty dep array?

Comment (sadan4) [reply src/plugins/imageZoom/components/Magnifier.tsx:64]: just looked up and [its not the same](https://react.dev/reference/react/useEffect#examples-dependencies)

however, if a dep array is passed, it doesn't work

Comment (Nuckyz) [reply src/plugins/imageZoom/components/Magnifier.tsx:64]: that must be why the layout effect doesn't work either

try a layout effect with no deps

**Pull request merged**